ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/MixedIcebergPartitionPlan.java

```diff
@@ -201,7 +212,10 @@ public boolean isFullNecessary() {
       if (!reachFullInterval()) {
         return false;
       }
-      return anyDeleteExist() || fragmentFileCount > getBaseSplitCount() || hasChangeFiles;
+      return anyDeleteExist()
+          || fragmentFileCount > getBaseSplitCount()
+          || rewriteSegmentFileCount + undersizedSegmentFileCount > getBaseSplitCount()
```

Comment:
The same as `MixedHivePartitionPlan`, I suggest removing this condition and sticking to the original code in this PR.

Reply:
Done.